### PR TITLE
adding "Conditional Variational Autoencoder for Sign Language Translation with Cross-Modal Alignment"

### PR DESCRIPTION
**Pull request by cleong110**
Making a PR to add http://arxiv.org/abs/2312.15645, one of the recent approaches (with code!) listed on https://github.com/ZechengLi19/Awesome-Sign-Language. 

I only intended to include f435dcadd968e357b75425fec5902a57dfb96180 in this PR, unfortunately, I made [the previous PR](https://github.com/sign-language-processing/sign-language-processing.github.io/pull/35) on my fork's master branch so this PR now includes those commits? I think in future if I make a new branch on my end for each commit that should prevent it. And I believe once the previous PR is merged this one should be mergeable easily as well?

### CODE REVIEW

**Comment by cleong110** [comment]
> Please pull and merge master into this branch. that should remove any changes from other PRs from the changelist. If not, please create a separate PR with only the relevant changes. (You should always branch out from master)

Can do

**Comment by cleong110** [comment]
That did it!

File: src/index.md [hunk 1]
```diff
@@ -726,6 +726,18 @@ The model features shared representations for different modalities such as text
 on several tasks such as video-to-gloss, gloss-to-text, and video-to-text. 
 The approach allows leveraging external data such as parallel data for spoken language machine translation.
 
+@zhaoConditionalVariationalAutoencoder2023 propose CV-SLT, an encoder-decoder based approach using conditional variational autoencoders.
```
Comment (AmitMY) [inline src/index.md:729]: note: i have separated each line after the period, for readability on git, and ease of commenting on specific lines. It changes nothing in the final format.

File: src/index.md [hunk 2]
```diff
@@ -726,6 +726,18 @@ The model features shared representations for different modalities such as text
 on several tasks such as video-to-gloss, gloss-to-text, and video-to-text. 
 The approach allows leveraging external data such as parallel data for spoken language machine translation.
 
+@zhaoConditionalVariationalAutoencoder2023 propose CV-SLT, an encoder-decoder based approach using conditional variational autoencoders. 
+In a nutshell, they run the video data through once with text and once without, and the model must learn how to correct for this.
```
Comment (AmitMY) [inline src/index.md:730]: This sentence is unclear. 
- "run the video through" what?
- "how to correct for this" - correct for what?

File: src/index.md [hunk 3]
```diff
@@ -726,6 +726,18 @@ The model features shared representations for different modalities such as text
 on several tasks such as video-to-gloss, gloss-to-text, and video-to-text. 
 The approach allows leveraging external data such as parallel data for spoken language machine translation.
 
+@zhaoConditionalVariationalAutoencoder2023 propose CV-SLT, an encoder-decoder based approach using conditional variational autoencoders. 
+In a nutshell, they run the video data through once with text and once without, and the model must learn how to correct for this. 
+They argue that the modality gap between video and text is a major issue, using the MMTLB method from [@chenSimpleMultiModalityTransfer2022a] to embed videos and text from PHOENIX2014T and quantify the modality gap, then showing that more similar embeddings tend to trend with better BLEU scores.
```
Comment (AmitMY) [inline src/index.md:731]: when introducing new acronyms, citing is not enough.
Either:
- "using the Multi-Modality Transfer Learning Baseline (MMTLB, @chenSimpleMultiModalityTransfer2022a)"
- or "using the baseline from @chenSimpleMultiModalityTransfer2022a"

Also note: you wrote "from [@...]" which will look like "from (Something et al.)" - you should remove the square brackets if you want to refer directly in the text.

Also, when referring to terms that were probably already in the text (PHOENIX2014T), Ideally, look them up quickly in the text. You will see that this one is mostly written "RWTH-PHOENIX-Weather-2014T", so for consistency, use that term

File: src/index.md [hunk 4]
```diff
@@ -726,6 +726,18 @@ The model features shared representations for different modalities such as text
 on several tasks such as video-to-gloss, gloss-to-text, and video-to-text. 
 The approach allows leveraging external data such as parallel data for spoken language machine translation.
 
+@zhaoConditionalVariationalAutoencoder2023 propose CV-SLT, an encoder-decoder based approach using conditional variational autoencoders. 
+In a nutshell, they run the video data through once with text and once without, and the model must learn how to correct for this. 
+They argue that the modality gap between video and text is a major issue, using the MMTLB method from [@chenSimpleMultiModalityTransfer2022a] to embed videos and text from PHOENIX2014T and quantify the modality gap, then showing that more similar embeddings tend to trend with better BLEU scores. 
+Their method to deal with this is a framework for training with two paths, which they call "prior" and "posterior".
```
Comment (AmitMY) [inline src/index.md:732]: this is not easily clear. anticident is a little far since last sentence is long. "with the modality gap"

**Review by AmitMY — CHANGES_REQUESTED** [review]
couldn't review the entire thing right now since i gotta go but left some notes. 
Please pull first.
I also advise that you use ChatGPT to feed in the entire text and it will make it clearer and tighter. Here is the style guide I use for this project for ChatGPT:
https://github.com/sign-language-processing/sign-language-processing.github.io?tab=readme-ov-file#style-guide

File: src/index.md (hunk 3 above)
Comment (cleong110) [reply src/index.md:731]: Good call on MMTLB acronym. 

PHOENIX2014T I was a little unsure about, so thanks for checking that. 

On that topic, is that the right citation? I did spend some time making sure that Zhao et al's citation and our citation both pointed to the same and correct paper. 
* In the "Gloss to pose" section I saw `Phoenix-14T dataset [@dataset:forster2014extensions]`, 
* but elsewhere I mostly see `@cihan2018neural`, so that's what I went to

**Comment by cleong110** [comment]
Thanks for all the helpful suggestions! Working on rewrites.

Re: "please pull first", I think it is in the correct state now, `cleong110:paper/CV-SLT` now shows as being synced up with `sign-language-processing:master`, is there more to do on that front?

File: src/index.md (hunk 3 above)
Comment (AmitMY) [reply src/index.md:731]: cihan2018neural is correct. just verified.

**Comment by cleong110** [comment]
Asked GPT4-o to help summarize the PDF of the paper for me, as advised. The conversation and prompt are at https://chat.openai.com/share/c720d444-a505-491e-b131-78e03e10e700. It actually did quite well, I think!

The explanation aligns with my understanding and seems to clarify. And seems to line up with Figure 2 here:
image

OK, so it seems the encoder is supposed to generate _distributions_ of possible embeddings, and the decoder is supposed to generate distributions of possible text translations. And the KL divergences are there to, like, pull the distributions with/without text inputs closer. 

So I guess the training process goes like:

1. Give the model visual input only. 
2. Encoder (some kind of vision transformer) takes that visual input and does the usual encoding stuff, making hidden states/embeddings. Those are the gray boxes
image
3. Encoder then passes that on to a Gaussian Network (I don't know the architecture of this one), and the GN generates a distribution of embeddings, the green boxes in this picture
image
4. Decoder takes the hidden states and the distributions from the Encoder and then outputs a distribution of possible translations, the orange boxes here
image
5. But then you do that whole thing, steps 2 through 4 again, but this time the model gets to include both input video and input text. Again you generate distributions for the embeddings, and distributions for the text translations
6. Now you've got your with-text and without-text distributions. KL divergences let you calculate how "far" apart those distributions are. You want to minimize these numbers. So you, presumably, use the KL divergences as a loss for backprop, I guess? Yeah looks like that's correct
image

So you've got three loss terms. One overall reconstruction loss, one for the encoder to help it align with-visual and without-visual distributions, and one for the decoder to help with consistency of outputs. 
1. $L_{CVAE}$ has a KL in it, it is presumably the one that works on consistency of embeddings/latent variables, as it deals with things like $q(z|x,y)$, which I read as "posterior distribution of latent variables given we have videos (x) and text (y). So this would be the "make ENCODER consistent with and without text" loss. This loss going down means that the "no text" encoder is generating latent variables more similar to the "with text" encoder's.
2. $L_{SD}$ Would then be the "DECODER consistent with and without text" loss. There's a KL divergence in there. This loss going down means that the full "no text" path is generating outputs more similar to the "with text" path.
3. $L_{AEP}$, then, has to be the overall reconstruction loss, for the prior path. This loss going down means that the prior path (without text) is getting better at generating the right translations.

So over time, these three losses go down. The encoder gets more consistent at encoding/embedding to the same thing internally with or without text. The decoder, given those embeddings, gets more consistent about generating the same output either way, with or without text. And the ability of the overall prior path to get to the right text gets better. 

I think I'm ready to rewrite it now.

File: src/index.md (hunk 1 above)
Comment (cleong110) [reply src/index.md:729]: I like that. Will do so going forward.

File: src/index.md (hunk 2 above)
Comment (cleong110) [reply src/index.md:730]: Rewrote significantly for clarity.

**Comment by cleong110** [comment]
Wrote a new summary myself, then [asked ChatGPT 3.5 to rewrite for conciseness.](https://chat.openai.com/share/486f383e-5f85-4fca-989c-8ec63d43e287)

A few minor edits to ChatGPT's version for style guide purposes gives us:

<code>@zhaoConditionalVariationalAutoencoder2023 introduces CV-SLT, employing conditional variational autoencoders to address the modality gap between video and text.
They assess the disparity using RWTH-PHOENIX-Weather-2014T data, correlating similar embeddings with improved BLEU scores.
Their approach involves guiding the model to encode visual and textual data similarly through two paths: one with visual data alone and one with both modalities.
Using KL divergences, they steer the model towards generating consistent embeddings and accurate outputs regardless of the path.
Once the model achieves consistent performance across paths, it can be utilized for translation without gloss supervision.
Evaluation on RWTH-PHOENIX-Weather-2014T [@cihan2018neural] and CSL-Daily [@dataset:huang2018video] datasets demonstrates its efficacy.
They provide a [code implementation](https://github.com/rzhao-zhsq/CV-SLT) based largely on @chenSimpleMultiModalityTransfer2022a.
</code>

Which I think ought to take care of all the issues:
- [x] sums it up well, 
- [x] fixes the consistency issue for RWTH-PHOENIX-Weather-2014T
- [x] should have the correct citation style, parentheses around the dataset citations and none around the ones that shouldn't have them
- [x] One sentence per line

**Comment by AmitMY** [comment]
Sorry, yes, I meant ask chatgpt to help with your writing. you write a draft, it gives tips to improve

File: src/index.md [hunk 5]
```diff
@@ -726,6 +726,16 @@ The model features shared representations for different modalities such as text
 on several tasks such as video-to-gloss, gloss-to-text, and video-to-text. 
 The approach allows leveraging external data such as parallel data for spoken language machine translation.
 
+@zhaoConditionalVariationalAutoencoder2023 introduce CV-SLT, employing conditional variational autoencoders to address the modality gap between video and text.
+They assess the disparity using RWTH-PHOENIX-Weather-2014T data, correlating similar embeddings with improved BLEU scores.
```
Comment (AmitMY) [inline src/index.md:730]: I think remove this line. If anything it should come at the end, but there is already a line on evaluation.
Also, this line does not read that well. everything else is good!

File: src/index.md [hunk 6]
```diff
@@ -726,6 +726,16 @@ The model features shared representations for different modalities such as text
...
+Their approach involves guiding the model to encode visual and textual data similarly through two paths: one with visual data alone and one with both modalities.
+Using KL divergences, they steer the model towards generating consistent embeddings and accurate outputs regardless of the path.
+Once the model achieves consistent performance across paths, it can be utilized for translation without gloss supervision.
+Evaluation on RWTH-PHOENIX-Weather-2014T [@cihan2018neural] and CSL-Daily [@dataset:huang2018video] datasets demonstrates its efficacy.
```
Comment (AmitMY) [inline src/index.md:734]: add "the":

Evaluation on `the` RWTH-PHOENIX-Weather-2014T [@cihan2018neural] and CSL-Daily [@dataset:huang2018video] datasets demonstrates its efficacy.

File: src/index.md [hunk 7]
```diff
@@ -726,6 +726,16 @@ The model features shared representations for different modalities such as text
...
+They provide a [code implementation](https://github.com/rzhao-zhsq/CV-SLT) based largely on @chenSimpleMultiModalityTransfer2022a.
+
```
Comment (AmitMY) [inline src/index.md:736]: if you want to keep this comment in the code, remove the new line before it, so it is clear it relates to this specific paragraph. otherwise, maybe someone will move the paragraph some day

File: src/references.bib [hunk 8]
```diff
@@ -2152,3 +2149,30 @@ @inproceedings{dataset:joshi-etal-2023-isltranslate
...
+@misc{zhaoConditionalVariationalAutoencoder2023,
+  title = {Conditional {{Variational Autoencoder}} for {{Sign Language Translation}} with {{Cross-Modal Alignment}}},
+  author = {Zhao, Rui and Zhang, Liang and Fu, Biao and Hu, Cong and Su, Jinsong and Chen, Yidong},
+  year = {2023},
+  month = dec,
+  number = {arXiv:2312.15645},
+  eprint = {2312.15645},
+  primaryclass = {cs},
+  publisher = {arXiv},
+  url = {http://arxiv.org/abs/2312.15645},
```
Comment (AmitMY) [inline src/references.bib:2162]: always search papers in semanticscholar.
For eample this paper: https://www.semanticscholar.org/search?q=Conditional%20%7B%7BVariational%20Autoencoder%7D%7D%20for%20%7B%7BSign%20Language%20Translation%7D%7D%20with%20%7B%7BCross-Modal%20Alignment%7D%7D&sort=relevance
was actually published in `  booktitle={AAAI Conference on Artificial Intelligence},`

arXiv citations are considered "bad"

**Review by AmitMY — CHANGES_REQUESTED** [review]
Overall looks good! minor corrections

File: src/references.bib (hunk 8 above)
Comment (cleong110) [reply src/references.bib:2162]: Ah, so it has! Good call.

Comment (cleong110) [reply src/references.bib:2162]: I went digging a bit, I think the "official" citation might be this one: https://ojs.aaai.org/index.php/AAAI/article/view/29937, 

which gives me the following bibtex:
```
@article{Zhao_Zhang_Fu_Hu_Su_Chen_2024, title={Conditional Variational Autoencoder for Sign Language Translation with Cross-Modal Alignment}, volume={38}, url={https://ojs.aaai.org/index.php/AAAI/article/view/29937}, DOI={10.1609/aaai.v38i17.29937}, abstractNote={Sign language translation (SLT) aims to convert continuous sign language videos into textual sentences. As a typical multi-modal task, there exists an inherent modality gap between sign language videos and spoken language text, which makes the cross-modal alignment between visual and textual modalities crucial. However, previous studies tend to rely on an intermediate sign gloss representation to help alleviate the cross-modal problem thereby neglecting the alignment across modalities that may lead to compromised results. To address this issue, we propose a novel framework based on Conditional Variational autoencoder for SLT (CV-SLT) that facilitates direct and sufficient cross-modal alignment between sign language videos and spoken language text. Specifically, our CV-SLT consists of two paths with two Kullback-Leibler (KL) divergences to regularize the outputs of the encoder and decoder, respectively. In the prior path, the model solely relies on visual information to predict the target text; whereas in the posterior path, it simultaneously encodes visual information and textual knowledge to reconstruct the target text. The first KL divergence optimizes the conditional variational autoencoder and regularizes the encoder outputs, while the second KL divergence performs a self-distillation from the posterior path to the prior path, ensuring the consistency of decoder outputs.We further enhance the integration of textual information to the posterior path by employing a shared Attention Residual Gaussian Distribution (ARGD), which considers the textual information in the posterior path as a residual component relative to the prior path. Extensive experiments conducted on public datasets demonstrate the effectiveness of our framework, achieving new state-of-the-art results while significantly alleviating the cross-modal representation discrepancy. The code and models are available at https://github.com/rzhao-zhsq/CV-SLT.}, number={17}, journal={Proceedings of the AAAI Conference on Artificial Intelligence}, author={Zhao, Rui and Zhang, Liang and Fu, Biao and Hu, Cong and Su, Jinsong and Chen, Yidong}, year={2024}, month={Mar.}, pages={19643-19651} }
```
which when I import to Zotero and out with BetterBibTex (to exclude e.g. the abstract, gives me 

```
@article{Zhao_Zhang_Fu_Hu_Su_Chen_2024,
  title = {Conditional Variational Autoencoder for Sign Language Translation with Cross-Modal Alignment},
  author = {Zhao, Rui and Zhang, Liang and Fu, Biao and Hu, Cong and Su, Jinsong and Chen, Yidong},
  year = {2024},
  month = mar,
  journal = {Proceedings of the AAAI Conference on Artificial Intelligence},
  volume = {38},
  number = {17},
  pages = {19643--19651},
  doi = {10.1609/aaai.v38i17.29937},
  url = {https://ojs.aaai.org/index.php/AAAI/article/view/29937},
  annotation = {Abstract note: Sign language translation (SLT) aims to convert continuous sign language videos into textual sentences. As a typical multi-modal task, there exists an inherent modality gap between sign language videos and spoken language text, which makes the cross-modal alignment between visual and textual modalities crucial. However, previous studies tend to rely on an intermediate sign gloss representation to help alleviate the cross-modal problem thereby neglecting the alignment across modalities that may lead to compromised results. To address this issue, we propose a novel framework based on Conditional Variational autoencoder for SLT (CV-SLT) that facilitates direct and sufficient cross-modal alignment between sign language videos and spoken language text. Specifically, our CV-SLT consists of two paths with two Kullback-Leibler (KL) divergences to regularize the outputs of the encoder and decoder, respectively. In the prior path, the model solely relies on visual information to predict the target text; whereas in the posterior path, it simultaneously encodes visual information and textual knowledge to reconstruct the target text. The first KL divergence optimizes the conditional variational autoencoder and regularizes the encoder outputs, while the second KL divergence performs a self-distillation from the posterior path to the prior path, ensuring the consistency of decoder outputs.We further enhance the integration of textual information to the posterior path by employing a shared Attention Residual Gaussian Distribution (ARGD), which considers the textual information in the posterior path as a residual component relative to the prior path. Extensive experiments conducted on public datasets demonstrate the effectiveness of our framework, achieving new state-of-the-art results while significantly alleviating the cross-modal representation discrepancy. The code and models are available at https://github.com/rzhao-zhsq/CV-SLT.}
}

```
@article{Zhao_Zhang_Fu_Hu_Su_Chen_2024,
  title = {Conditional Variational Autoencoder for Sign Language Translation with Cross-Modal Alignment},
  author = {Zhao, Rui and Zhang, Liang and Fu, Biao and Hu, Cong and Su, Jinsong and Chen, Yidong},
  year = {2024},
  month = mar,
  journal = {Proceedings of the AAAI Conference on Artificial Intelligence},
  volume = {38},
  number = {17},
  pages = {19643--19651},
  doi = {10.1609/aaai.v38i17.29937},
  url = {https://ojs.aaai.org/index.php/AAAI/article/view/29937}
}
... which I will edit the references.bib to

File: src/index.md (hunk 5 above)
Comment (cleong110) [reply src/index.md:730]: Yeah I was iffy on "assess the disparity" as well.

**Comment by cleong110** [comment]
How do you typically prompt ChatGPT for writing improvement suggestions, btw? I just asked it to help me rewrite for conciseness, being sure to mention it was a summary of an academic paper

**Review by AmitMY — APPROVED** [review]
job well done
